File: arena-xunit/src/Playbook/ActiveOraclePlaybook.cs
using System;
using ArenaDotnet.Xunit.Ffi;
using ArenaDotnet.Xunit.Support;
using Newtonsoft.Json;

namespace ArenaDotnet.Xunit.Playbook;

public sealed class ActiveOraclePlaybook : ActivePlaybook
{
    public ActiveOraclePlaybook(IntPtr handle) : base(handle) { }

    public void Verify(string query, int expectedValue)
    {
        var spec = ArenaJson.Serialize(new OracleVerifySpec
        {
            Query = query,
            ExpectedValue = expectedValue,
        });
        ArenaBindings.OraclePlaybookVerify(_handle, spec);
    }

    [JsonObject(ItemNullValueHandling = NullValueHandling.Ignore)]
    private sealed class OracleVerifySpec
    {
        [JsonProperty("query")] public string? Query { get; set; }
        [JsonProperty("expected_value")] public int ExpectedValue { get; set; }
    }
}

File: arena-junit/src/main/java/arena/junit/playbook/oracledb/ActiveOraclePlaybook.java
package arena.junit.playbook.oracledb;

import arena.junit.ffi.ArenaBindingError;
import arena.junit.ffi.ArenaBindings;
import arena.junit.playbook.ActivePlaybook;
import arena.junit.support.ArenaJson;

import com.fasterxml.jackson.databind.node.ObjectNode;
import com.sun.jna.Pointer;

public class ActiveOraclePlaybook extends ActivePlaybook {
  protected ActiveOraclePlaybook(Pointer handle) {
    super(handle);
  }

  public void verify(String query, int expectedValue) {
    ObjectNode spec = ArenaJson.object();
    spec.put("query", query);
    spec.put("expected_value", expectedValue);
    try {
      ArenaBindings.oraclePlaybookVerify(handle(), ArenaJson.MAPPER.writeValueAsString(spec));
    } catch (ArenaBindingError e) {
      noteBodyFailure();
      throw e;
    } catch (Exception e) {
      throw new ArenaBindingError(e.getMessage());
    }
  }
}
